utils/utils.go

```diff
@@ -224,3 +227,114 @@ func MovePauseProcessToScope(pausePidPath string) {
 		}
 	}
 }
+
+// TransferRootless creates new podman processes using exec.Command and sudo, transferring images between the given source and destination users
+func TransferRootless(source entities.ImageScpOptions, dest entities.ImageScpOptions, podman string, parentFlags []string) error {
+	saveCommand := []string{"save", "--output", source.File, source.Image}
+	loadCommand := []string{"load", "--input", dest.File}
+
+	cmd := util.CreateSCPCommand(podman, parentFlags, saveCommand, true)
+	if source.User == "root" {
+		cmd.Args = append([]string{"sudo"}, cmd.Args...)
+	}
+	logrus.Debug("Executing save command")
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	cmd = util.CreateSCPCommand(podman, parentFlags, loadCommand, true)
+	if source.User != "root" {
+		cmd.Args = append([]string{"sudo"}, cmd.Args...)
+	}
+	logrus.Debug("Executing load command")
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// TransferRootless creates new podman processes using exec.Command and su/machinectl, transferring images between the given source and destination users
+func TransferRootful(source entities.ImageScpOptions, dest entities.ImageScpOptions, podman string, parentFlags []string) error {
+	saveCommand := []string{"save", "--output", source.File, source.Image}
+	loadCommand := []string{"load", "--input", dest.File}
+	// if executing using sudo or transferring between two users, the TransferRootless approach will not work, default to using machinectl or su as necessary.
+	// the approach using sudo is preferable and more straightforward. There is no reason for using sudo in these situations
+	// since the feature is meant to transfer from root to rootless an vice versa without explicit sudo evocaiton.
+	var uSave *user.User
+	var uLoad *user.User
+	var err error
+	source.User = strings.Split(source.User, ":")[0] // split in case provided with uid:gid
+	dest.User = strings.Split(dest.User, ":")[0]
+	uSave, err = lookupUser(source.User)
+	if err != nil {
+		return err
+	}
+	switch {
+	case dest.User != "": // if we are given a destination user, check that first
+		uLoad, err = lookupUser(dest.User)
+		if err != nil {
+			return err
+		}
+	case uSave.Name != "root": // else if we have no destination user, and source is not root that means we should be root
+		uLoad, err = user.LookupId("0")
+		if err != nil {
+			return err
+		}
+	default: // else if we have no dest user, and source user IS root, we want to be the default user.
+		uString := os.Getenv("SUDO_USER")
+		if uString == "" {
+			uLoad, err = user.LookupId("1000")
```

Comment:
WHy are you hard coding UID 1000?  That is not my UID?

Reply:
changed to an error